internal/services/privatednsresolver/private_dns_resolver_dns_forwarding_ruleset_data_source.go

```diff
@@ -0,0 +1,111 @@
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/dnsresolver/2022-07-01/dnsforwardingrulesets"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tags"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type PrivateDNSResolverDnsForwardingRulesetDataSourceModel struct {
+	Name                         string            `tfschema:"name"`
+	ResourceGroupName            string            `tfschema:"resource_group_name"`
+	DnsResolverOutboundEndpoints []string          `tfschema:"private_dns_resolver_outbound_endpoint_ids"`
+	Location                     string            `tfschema:"location"`
+	Tags                         map[string]string `tfschema:"tags"`
+}
+
+type PrivateDNSResolverDnsForwardingRulesetDataSource struct{}
+
+var _ sdk.DataSource = PrivateDNSResolverDnsForwardingRulesetDataSource{}
+
+func (r PrivateDNSResolverDnsForwardingRulesetDataSource) ResourceType() string {
+	return "azurerm_private_dns_resolver_dns_forwarding_ruleset"
+}
+
+func (r PrivateDNSResolverDnsForwardingRulesetDataSource) ModelObject() interface{} {
+	return &PrivateDNSResolverDnsForwardingRulesetDataSourceModel{}
+}
+
+func (r PrivateDNSResolverDnsForwardingRulesetDataSource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return dnsforwardingrulesets.ValidateDnsForwardingRulesetID
+}
+
+func (r PrivateDNSResolverDnsForwardingRulesetDataSource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+	}
+}
+
+func (r PrivateDNSResolverDnsForwardingRulesetDataSource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"location": commonschema.LocationComputed(),
+
+		"private_dns_resolver_outbound_endpoint_ids": {
+			Type:     pluginsdk.TypeList,
+			Computed: true,
+			Elem: &pluginsdk.Schema{
+				Type: pluginsdk.TypeString,
+			},
+		},
+
+		"tags": tags.SchemaDataSource(),
+	}
+}
+
+func (r PrivateDNSResolverDnsForwardingRulesetDataSource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.PrivateDnsResolver.DnsForwardingRulesetsClient
+
+			var state PrivateDNSResolverDnsForwardingRulesetDataSourceModel
+			if err := metadata.Decode(&state); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			var top int64 = 1
+			resp, err := client.ListByResourceGroupCompleteMatchingPredicate(ctx,
```

Comment:
Is there a reason we are getting this resource via a list rather than grabbing it directly?

Reply:
done